### PR TITLE
Handle 6 decimal places cloud event

### PR DESCRIPTION
**Pull request by rakshith91**
Fixes https://github.com/Azure/azure-sdk-for-python/issues/18602

### CODE REVIEW

File: sdk/core/azure-core/azure/core/_utils.py [hunk 1]
```diff
@@ -53,6 +53,17 @@ def _convert_to_isoformat(date_time):
...
+    check_decimal = timestamp.split('.')
+    if len(check_decimal) > 1:
+        decimal_str = ""
+        for digit in check_decimal[1]:
+            if digit.isdigit():
+                decimal_str += digit
+            else:
+                break
+        if len(decimal_str) > 6:
+            timestamp = timestamp.replace(decimal_str, decimal_str[0:6])
```
Comment (xiangyan99) [inline sdk/core/azure-core/azure/core/_utils.py:65]: I think we have this discussion. If the precision is more than 6, should we just drop the rest?

Comment (rakshith91) [reply sdk/core/azure-core/azure/core/_utils.py:65]: yes - that's what we have decided :)

Comment (xiangyan99) [reply sdk/core/azure-core/azure/core/_utils.py:65]: Do we want to log a warning message for precision loss?

Comment (rakshith91) [reply sdk/core/azure-core/azure/core/_utils.py:65]: i don't think it's necessary -  this is an event, there is no scenario where you would re-create a PUT request for ARM and expect precise value.

Comment (xiangyan99) [reply sdk/core/azure-core/azure/core/_utils.py:65]: and is it possible that timestamp.replace(decimal_str, decimal_str[0:6]) applies to some other places other than the last segment?

Comment (rakshith91) [reply sdk/core/azure-core/azure/core/_utils.py:65]: nope - it should not

**Review by xiangyan99 — APPROVED** [review]

**Pull request merged**